### PR TITLE
remove hardcoded list of resources

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:

```release-note
NONE
```

Removes the hardcoded list of resources in cmdutil.ValidResourceTypeList (which was not being kept up to date) and instead suggests using the `kubectl api-resources` command in order to retrieve a discovery-based list of supported resources.

I prefer this approach over updating `cmdutil.ValidResourceTypeList` to be based on discovery in order to avoid potential calls to the server while building the help output of commands.

cc @soltysh

### CODE REVIEW

**Comment by mengqiy** [comment]
IMO redirect the users to use api-resources is a good idea. It mitigates the long-standing problem that kubectl resources list is always out-of-date.
@shiywang Do you have bandwidth to review this?
/assign shiywang

File: pkg/kubectl/cmd/describe.go [hunk 1]
```diff
@@ -69,10 +70,38 @@ var (
 		kubectl describe pods frontend`))
 )
 
-func NewCmdDescribe(f cmdutil.Factory, out, cmdErr io.Writer) *cobra.Command {
-	options := &resource.FilenameOptions{}
-	describerSettings := &printers.DescriberSettings{
-		ShowEvents: true,
+type DescribeOptions struct {
```
Comment (soltysh) [inline pkg/kubectl/cmd/describe.go]: Can this change be a separate PR, if not at least a separate commit for ease of review.

**Review by soltysh — CHANGES_REQUESTED** [review]
One nit, mostly due to readability and this will be good to go.

File: pkg/kubectl/cmd/describe.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/kubectl/cmd/describe.go]: Done

File: pkg/kubectl/cmd/describe.go [hunk 2]
```diff
@@ -67,10 +68,36 @@ var (
...
+	CmdParent string
+	Selector  string
+	Namespace string
+
+	Describer func(*meta.RESTMapping) (printers.Describer, error)
+	Builder   func() *resource.Builder
```
Comment (soltysh) [inline pkg/kubectl/cmd/describe.go]: It's better to just have this be the `Builder` not a function returning one.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve no-issue

**Review by soltysh — APPROVED** [review]
/lgtm
